### DISCUSSION TITLE
SyntaxError: Unexpected token ';'

### DISCUSSION THREAD

**Discussion by joelparkerhenderson** [alpinejs/alpine#1409] [7. Bugs (archived)]
I'm using Alpine, and for the first time I'm trying it via webpack and node, yet get this error:

```sh
   webpack-internal:///./node_modules/alpinejs/dist/alpine.js 113:11 Uncaught SyntaxError: Unexpected token ';'
```

Package lock seems to be current:

```
"resolved": "https://registry.npmjs.org/alpinejs/-/alpinejs-2.7.3.tgz",
```

What I'm learning is that this may be a babel related issue, and/or an issue with something incorrectly loading alpinejs as an HTML file rather than as a JavaScript file.

The HTML head includes a typical script tag:

```html
<script defer type="text/javascript" src="/js/app.js"></script>
```

The relevant web page does actually load in a typical browser.

In case it's relevant, I'm using Alpine in a typical new Elixir Phoenix project, using the typical default webpack.config:

```js
const path = require('path');
const glob = require('glob');
const HardSourceWebpackPlugin = require('hard-source-webpack-plugin');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const TerserPlugin = require('terser-webpack-plugin');
const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
const CopyWebpackPlugin = require('copy-webpack-plugin');

module.exports = (env, options) => {
  const devMode = options.mode !== 'production';

  return {
    optimization: {
      minimizer: [
        new TerserPlugin({ cache: true, parallel: true, sourceMap: devMode }),
        new OptimizeCSSAssetsPlugin({})
      ]
    },
    entry: {
      'app': glob.sync('./vendor/**/*.js').concat(['./js/app.js'])
    },
    output: {
      filename: '[name].js',
      path: path.resolve(__dirname, '../priv/static/js'),
      publicPath: '/js/'
    },
    devtool: devMode ? 'eval-cheap-module-source-map' : undefined,
    module: {
      rules: [
        {
          test: /\.js$/,
          exclude: /node_modules/,
          use: {
            loader: 'babel-loader'
          }
        },
        {
          test: /\.[s]?css$/,
          use: [
            MiniCssExtractPlugin.loader,
            'css-loader',
            'sass-loader',
          ],
        },
        {
          test: /\.(eot|otf|ttf|woff|woff2)$/,
          use: [
            {
              loader: 'file-loader',
              options: { 
                name: '[name].[ext]', 
                outputPath: '../fonts' 
              }
            },
          ],
        }
      ]
    },
    plugins: [
      new MiniCssExtractPlugin({ filename: '../css/app.css' }),
      new CopyWebpackPlugin([{ from: 'static/', to: '../' }])
    ]
    .concat(devMode ? [new HardSourceWebpackPlugin()] : [])
  }
};
```

Guidance much appreciated, thank you!

**Comment by HugoDF** [comment]
Hey @joelparkerhenderson have you got a minimum reproduction case repository you can share?

It's going to be very difficult to debug a build issue without the full build config code.

**Comment by joelparkerhenderson** [comment]
Yes thank you. 

When I try loading via a typical script tag and a CDN, rather than via npm, webpack, etc.:

```html
<body>
    <script src="https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.7.3/dist/alpine.min.js" defer></script>
```

Then the error message is:

```sh
https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.7.3/dist/alpine.min.js 6:1783 Uncaught SyntaxError: Unexpected token ';'
```

For comparison, when I try loading via GitHub master, then it works fine:

```html
<body>
    <script src="https://raw.githubusercontent.com/alpinejs/alpine/master/dist/alpine.js" defer></script>
```

When I try loading via NPM, the browser shows these errors:

```
Uncaught SyntaxError: missing } after property list
    saferEval alpine.js:118
    Component alpine.js:1423
    initializeComponent alpine.js:1870
    start alpine.js:1815
    discoverComponents alpine.js:1829
    discoverComponents alpine.js:1828
    start alpine.js:1814
    async* alpine.js:1902
    <anonymous> alpine.js:2
    <anonymous> alpine.js:5
```

**Comment by SimoTod** [comment]
That error suggests you have a typo in your markup, can you post the component?

**Comment by joelparkerhenderson** [comment]
Yes thank you and I'll simplify it. 

Is it accurate to say that the errors above are potentially because the page HTML/JS has problems that are due to me, rather than the errors above being within Alpine itself?

Perhaps another way to ask... is Alpine HTML/JS doing some kind of processing akin to a "eval" inside of the Alpine script, such that what I wrote on the page would be causing the errors to seem to be coming from inside of Alpine?

If so, is there a recommended way that I can process my page, such as a linter or preprocessor, that could potentially help verify the page is written correctly such that it's valid input to Alpine?

If there's a way to donate to you or your favorite charity, please let me know. I greatly appreciate your help!

**Comment by HugoDF** [comment]
Alpine uses a sort of `eval` to run the contents of directives (`x-data`, `x-on:click`) etc.

So yes, if you've got invalid JavaScript in directives, an error with be thrown that looks like it's coming from Alpine but is actually caused by the code being evaluated.

There's a PR and another issue to address this.

No Linters are available at the moment.

**Comment by HugoDF** [comment]
@joelparkerhenderson with the latest Alpine.js release, you should get a warning with the expression/element at which the syntax error is happening

**Comment by Heiss** [comment]
Hello,

i am trying to use alpine in combination with webpack and babel, but i am getting "unexpected token" for this line: "@click.away="closeSideMenu". I tried to figure out, how to configure it right. Is there somewhere an example or  tutorial to do so? Or is it not possible to work together?
My package.json looking very similar to the first issuer.

Thank you.

**Comment by HugoDF** [comment]
> Hello,
> 
> i am trying to use alpine in combination with webpack and babel, but i am getting "unexpected token" for this line: "@click.away="closeSideMenu". I tried to figure out, how to configure it right. Is there somewhere an example or  tutorial to do so? Or is it not possible to work together?
> My package.json looking very similar to the first issuer.
> 
> Thank you.

Is the issue with the `@`? Does the same happen if you use `x-on:click.away` instead?

**Comment by trev-dev** [comment]
This error's pretty ambiguous and threw me for a loop. If AlpineJS doesn't appreciate what you've written in your markup then you need to test listeners until you find the "nearest broken event" to a chain of broken event listeners and figure it out.

In my case my markup didn't expect a key name for an object to contain spaces, thanks to user input, and it ended up being a broken object reference, I.E. `favoritePie.Pie Flavor` instead of `favoritePie['Pie Flavor']`

**Reply by SimoTod** [reply]
The console should also show you a warning printing out the directive and the dom element that caused the issue (you can also right click and jump to the element in the inspect tab). Was it not the case for you?
Those errors are not strictly generated by Alpine but are just js error caused by bad syntax so it's not immediate to reword them unless you add a lot of syntax check to prevent user from writing invalid statements (but then the other half of the users will complain because the library is getting too big).

**Comment by heshamelmasry77** [comment]
so what is the solution for this now?

**Comment by f-liva** [comment]
Same issue here
